api/src/app/modules/submission/submission.service.ts

```diff
@@ -664,15 +667,66 @@ export class SubmissionService extends DataService<Submission, Repository<Submis
       await this.updateProjectLocation(submission.projectId, user); // This will set geometry_latlong to null.
     }
     else {
-      await this.saveAndUpdateSpatialSubmission(submission, spatialObjectCode, user);
+      await this.repository.save(submission);
+      await this.updateProjectLocationAndSpatialAreaOrLength(submission, spatialObjectCode, user)
     }
   }
 
-  private async saveAndUpdateSpatialSubmission(
+  private getSpatialTableNameBySpatialObjectCode(spatialObjectCode: SpatialObjectCodeEnum) {
+    let spatialTableName: string; 
+    switch (spatialObjectCode) {
+      case SpatialObjectCodeEnum.CUT_BLOCK:
+        spatialTableName = CutBlock.tableName
+        break;
+      case SpatialObjectCodeEnum.ROAD_SECTION:
+        spatialTableName = RoadSection.tableName;
+        break;
+      case SpatialObjectCodeEnum.WTRA:
+        spatialTableName = RetentionArea.tableName;
+        break;
+      default:
+        throw new BadRequestException("Unrecognized spatial object code.");
+    }
+    return spatialTableName;
+  }
+
+  /**
+   * Simplify the geometry using Douglas-Peucker algorithm with a tolerance of 2.5m,
+   * which is established by Forestry Geospatial experts as the standard accuracy level for forestry applications.
+   * @param submissionID the id for the FOM spatial submission.
+   * @param spatialObjectCode the GeoSpatial Object Type for this spatial submission.
+   * @returns no return. Update and simplify the 'geometry' on particular spatial object type table.
+   */
+  private async simplifyGeometry(submissionID: number, spatialObjectCode: SpatialObjectCodeEnum) {
+    this.logger.debug(`Simplify geometry for submittion: ${submissionID}, spatial object type: ${spatialObjectCode}`);
+
+    const buildUpdateQuery = (submissionID: number, spatialObjectCode: SpatialObjectCodeEnum) => {
+      let targetSpatialTable = this.getSpatialTableNameBySpatialObjectCode(spatialObjectCode);
+      return `
+        UPDATE app_fom.${targetSpatialTable} SET geometry=ST_SimplifyPreserveTopology(geometry, 2.5) where submission_id = ${submissionID};
+      ` 
+    }
```

Comment:
We can certainly combine this into query below line like before if wish to embed the query statement.